staging/src/kubevirt.io/api/migrations/v1alpha1/types.go

```diff
@@ -0,0 +1,196 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package v1alpha1
+
+import (
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	k6tv1 "kubevirt.io/api/core/v1"
+)
+
+// MigrationPolicy holds migration policy (i.e. configurations) to apply to a VM or group of VMs
+//
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+// +genclient
+// +genclient:nonNamespaced
+type MigrationPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              MigrationPolicySpec `json:"spec" valid:"required"`
+	// +nullable
+	Status MigrationPolicyStatus `json:"status,omitempty"`
+}
+
+type MigrationPolicySpec struct {
+	Selectors *Selectors `json:"selectors"`
+
+	//+optional
+	AllowAutoConverge *bool `json:"allowAutoConverge,omitempty"`
+	//+optional
+	BandwidthPerMigration *resource.Quantity `json:"bandwidthPerMigration,omitempty"`
+	//+optional
+	CompletionTimeoutPerGiB *int64 `json:"completionTimeoutPerGiB,omitempty"`
```

Comment:
should this be a `uint64`?

Reply:
It's `int64` for KubevirtCR's [MigrationConfiguration](https://kubevirt.io/api-reference/master/definitions.html#_v1_migrationconfiguration) so I went with the same one